src/logics/ArithLogic.cc

```diff
@@ -529,14 +545,18 @@ PTRef ArithLogic::mkNeg(PTRef tr) {
         PTRef tr_n = mkFun(symref, std::move(args));
         return tr_n;
     }
-    if (isTimes(symref)) { // constant * var-like
+    if (isTimesLin(symref)) { // constant * (var-like \/ times-nonlin)
         assert(getPterm(tr).size() == 2);
-        auto [var, constant] = splitTermToVarAndConst(tr);
+        auto [var, constant] = splitPolyTerm(tr);
         return constant == getMinusOneForSort(getSortRef(symref)) ? var : mkFun(symref, {var, mkNeg(constant)});
     }
-    if (isNumVarLike(symref)) {
+    if (isTimesNonlin(symref)) {
+        SRef returnSort = getSortRef(tr);
+        return mkFun(getTimesLinForSort(returnSort), {tr, getMinusOneForSort(returnSort)});
+    }
+    if (isMonomial(symref)) {
```

Comment:
These two cases are identical, no?
`isMonomial` is sufficient.

Reply:
Fixed